packages/datetime/test/datePickerTests.tsx

```diff
@@ -26,6 +30,139 @@ describe("<DatePicker>", () => {
+            const selectedFirstDay = 3;
+            const wrapper = mount(<DatePicker dayPickerProps={{ firstDayOfWeek: selectedFirstDay }} />);
+            const firstWeekday = wrapper.find("Weekday").first();
+            assert.equal(firstWeekday.prop("weekday"), selectedFirstDay);
+        });
+
+        it("doesnt show outside days if specified", () => {
+            const defaultValue = new Date(2017, Months.SEPTEMBER, 1, 12);
+            const wrapper = mount(
+                <DatePicker defaultValue={defaultValue} dayPickerProps={{ enableOutsideDays: false }} />,
+            );
+
+            const dayIsHidden = (day: any): boolean =>
+                day.prop("empty") && !day.prop("ariaSelected") && day.prop("ariaDisabled");
+
+            assert.isTrue(dayIsHidden(wrapper.find("Day").at(0)));
+            assert.isTrue(dayIsHidden(wrapper.find("Day").at(1)));
+            assert.isTrue(dayIsHidden(wrapper.find("Day").at(2)));
+            assert.isTrue(dayIsHidden(wrapper.find("Day").at(3)));
+            assert.isTrue(dayIsHidden(wrapper.find("Day").at(4)));
+
+            assert.isFalse(dayIsHidden(wrapper.find("Day").at(5)));
+        });
+
+        it("respects default of true for enableOutsideDays prop", () => {
+            const defaultValue = new Date(2017, Months.SEPTEMBER, 1);
+            const firstDayInView = new Date(2017, Months.AUGUST, 27, 12, 0);
+            const wrapper = mount(<DatePicker defaultValue={defaultValue} />);
+            const firstDay = wrapper.find("Day").first();
+            assert.isTrue(areDatesEqual(new Date(firstDay.prop("day")), firstDayInView));
+        });
+
+        it("disables days as specified in disableDays daypicker prop in addition to min/max", () => {
+            const defaultValue = new Date(2017, Months.SEPTEMBER, 1);
+            const disableFridays = { daysOfWeek: [5] };
+            const { getDay } = wrap(
+                <DatePicker
+                    defaultValue={defaultValue}
+                    maxDate={new Date(2017, Months.SEPTEMBER, 20)}
+                    dayPickerProps={{ disabledDays: disableFridays }}
+                />,
+            );
+            assert.isTrue(getDay(15).hasClass(Classes.DATEPICKER_DAY_DISABLED));
+
+            assert.isTrue(getDay(21).hasClass(Classes.DATEPICKER_DAY_DISABLED));
+            assert.isFalse(getDay(10).hasClass(Classes.DATEPICKER_DAY_DISABLED));
+        });
+
+        it("respects default disabledDays value", () => {
+            const defaultValue = new Date(2017, Months.SEPTEMBER, 1);
+            const { getDay } = wrap(
+                <DatePicker defaultValue={defaultValue} maxDate={new Date(2017, Months.SEPTEMBER, 20)} />,
+            );
+            assert.isTrue(getDay(21).hasClass(Classes.DATEPICKER_DAY_DISABLED));
+            assert.isFalse(getDay(10).hasClass(Classes.DATEPICKER_DAY_DISABLED));
+        });
+
+        it("prefers blueprint props over dayPickerProps", () => {
```

Comment:
If we let `locale`, `localeUtils`, and `modifiers` be overwritten per my comment way above, this test will need to change. I'd also prefer a more explicit description, e.g.

`allows top-level locale, localeUtils, and modifiers to be overridden by same props in dayPickerProps`